packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -216,6 +216,15 @@ export class InternalModelManager {
     this.capitalizedApiIdentifier = camelize(apiIdentifier);
   }
 
+  private getRecordFromData(record: RecordData) {
+    let recordData = record;
+    if (this.apiIdentifier in record) {
```

Comment:
> The only exception is if there is a field with the same api identifier as the model, then the params must be wrapped in the model api identifier; this is because there's no way to tell if the consumer is passing in model params or field value.

If we wanted one heuristic, we could check to see if there's only a single key that's the API identifier and is an object. It's also not perfect, in the case of JSON fields, but I think could be an added safety measure.

If we wanted to prevent any issues we could make it a problem to have a field with the same API identifier as its model. Worth doing that? If so, this PR or another?

Reply:
I think that makes sense considering there aren't any exported params for the internal api

Reply:
I think that a model with a field that has the same name will actually be a somewhat regularly occurring thing -- I feel like people do that all the time where they make an `Image` model with an `image` File field or what have you. I think its a bit overbearing to say that is a problem and that they should change their design because we have an ambiguity elsewhere. Does the heuristic still work if we don't make that a problem?

Reply:
> Does the heuristic still work if we don't make that a problem?

It doesn't in the case that I mentioned where the field is a JSON type, then we do have the potential for ambiguity (which I should have clarified is what I envisioned the problem could be scoped to).

I suspect this case is probably rare enough that a problem wouldn't be overbearing. In most cases it would probably end up in a failed API call because the nested schema likely doesn't match the model's schema, but here's the rare case I'm thinking of where both styles work:

```js
// `Event` model with fields:
//   - event: JSON (optional)
//   - name: String (optional)
//   - type: String (default = "unknown")

await api.internal.create({ event: { type: "test" });
await api.internal.create({ event: { event: { type: "test" } });
```